cpp/src/streaming/cudf/partition.cpp

```diff
@@ -24,14 +25,15 @@ Node partition_and_pack(
 
     co_await ctx->executor()->schedule();
     while (true) {
-        auto table = co_await ch_in->receive_or(nullptr);
-        if (table == nullptr) {
+        auto msg = co_await ch_in->receive();
+        if (msg.empty()) {
             break;
         }
+        auto table = msg.release<TableChunk>();
         auto reservation = ctx->br()->reserve_and_spill(
-            MemoryType::DEVICE, table->make_available_cost(), false
+            MemoryType::DEVICE, table.make_available_cost(), false
         );
-        auto tbl = table->make_available(reservation, table->stream());
+        auto tbl = table.make_available(reservation, table.stream());
```

Comment:
*question*: Since the data in a message is now stored in a shared pointer, does this now potentially make a _copy_ of the `table` (since someone else might also have the same message)?